### ISSUE TITLE
Cannot set Timestamp and UsernameToken order

### ISSUE THREAD

**Issue by duanemaxwell** [savonrb/akami#23]
I've run into a situation where a server's documentation specifies that the Timestamp node must appear before the UsernameToken node, or more precisely, that the Timestamp node must be first.  The output I'm getting from Savon/Akami is UsernameToken, then Timestamp.

Much to my surprise, this is apparently part of the WSSE spec (6.7 [Security Header Layout]) - that you can require that the Timestamp node be first or last in the Security node.  Unfortunately, it seems this library uses a hash, so that the order of the items when converted to XML is not guaranteed.  My understanding is that there are several SOAP server implementations that require a particular order.

**Comment by duanemaxwell** [comment]
Here's my monkey patch to fix my immediate problem:

```ruby
module Akami
  class WSSE
    def to_xml
      if signature? and signature.have_document?
        Gyoku.xml wsse_signature.merge!(hash)
      elsif username_token? && timestamp?
        sec = wsse_username_token.merge!(wsu_timestamp) {
          |key, v1, v2| v1.merge!(v2) {
            |key, v1, v2| v1.merge!(v2)
          }
        }
        sec['wsse:Security'].merge! :order! =>['wsu:Timestamp','wsse:UsernameToken']
        Gyoku.xml sec
      elsif username_token?
        Gyoku.xml wsse_username_token.merge!(hash)
      elsif timestamp?
        Gyoku.xml wsu_timestamp.merge!(hash)
      else
        ""
      end
    end
  end
end
```

**Comment by pcai** [comment]
should be fixed in #24
